expression/builtin_time.go

```diff
@@ -3030,6 +3054,90 @@ func (du *baseDateArithmetical) vecGetDateFromInt(b *baseBuiltinFunc, input *chu
 	return nil
 }
 
+func (du *baseDateArithmetical) vecGetDateFromReal(b *baseBuiltinFunc, input *chunk.Chunk, unit string, result *chunk.Column) error {
+	n := input.NumRows()
+	buf, err := b.bufAllocator.get()
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf)
+	if err := b.args[0].VecEvalReal(b.ctx, input, buf); err != nil {
+		return err
+	}
+
+	result.ResizeTime(n, false)
+	result.MergeNulls(buf)
+	dates := result.Times()
+	f64s := buf.Float64s()
+	sc := b.ctx.GetSessionVars().StmtCtx
```

Comment:
Same as above, Session context should not be needed in ParseTimeFromFloat64.

Reply:
Ditto. Will do.

Reply:
Addressed.

Reply:
Again, sorry for this...